### PR TITLE
Rework Fake.Core.Target package

### PR DESCRIPTION
**Pull request by devcrafting**
**/!\ There is a breaking change with NuGet package rename to be consistent between Nuget package and namespace

- Documentation improved (I reordered paragraphs to put important notes first)
- Add more Obsolete messages in TargetHelper and AdditionalSyntax (including 2 new codes for "not yet migrated" and "no more accessible")
- Remove remaning "Target" suffix
- Rename Print in ListAvailable

### CODE REVIEW

**Comment by matthid** [comment]
Apparently this needs a merge/rebase after the latest PR merges, sorry.

File: help/markdown/fake-migrate-to-fake-5.md [hunk 1]
```diff
@@ -13,30 +13,34 @@ This "new" version has several advantages:
 * Paket bootstrapper / build.cmd and build.sh are no longer required (you can still use them)
 * This will be the only Version available in FAKE 6
 
-Therefore you have the FAKE 5 timeframe to update your build scripts to the new version.
+Therefore you have the FAKE 5 timeframe to update your build scripts to the new version. If you have any issues in the migration process, please see [how to fill issues or discuss about your issues](/contributing.html) (don't be shy about contributing ;)).
 
 ## Migration Guide
 
-Upgrading to FAKE 5 is a multi step process and has various manual steps in between. Here are the steps:
+Upgrading to FAKE 5 is a multi step process and has various manual steps in between. **If you do these steps out of order it will be a lot harder for you to migrate the script successfully**. Here are the steps:
```
Comment (matthid) [inline help/markdown/fake-migrate-to-fake-5.md]: Good point.

File: src/app/Fake.Core.Target/Target.fs [hunk 2]
```diff
@@ -407,14 +407,6 @@ module Target =
     /// [omit]
     let internal isListMode = Environment.hasEnvironVar "list"
 
-    /// Prints all available targets.
-    let internal listTargets() =
-        Trace.tracefn "Available targets:"
-        getTargetDict().Values
-          |> Seq.iter (fun target ->
-                Trace.tracefn "  - %s %s" target.Name (match target.Description with Some d -> " - " + d | _ -> "")
-                Trace.tracefn "     Depends on: %A" target.Dependencies)
```
Comment (matthid) [inline src/app/Fake.Core.Target/Target.fs]: Nice catch.

**Comment by devcrafting** [comment]
Rebase done (someone already fixed doc around Target)

**Comment by devcrafting** [comment]
Any chance to have a look at this PR @matthid ? Thanks

**Comment by matthid** [comment]
Yes I'm kind of slow at the moment. next step is to release all the stuff I've merged and update to netcoreapp20 and fixing the release pipeline (some things are not yet automated - see fake.api.github pr). Than I can test the release pipeline for example with this pr

File: build-web-bundles.fsx [hunk 3]
```diff
@@ -57,4 +57,4 @@ Target "Default" DoNothing
     ==> "Default"
 
 // start build
-RunParameterTargetOrDefault "target" "Default"
+RunTargetOrDefault "Default"
```
Comment (matthid) [inline build-web-bundles.fsx]: `Target.RunParameterOrDefault` or `Target.RunOrDefault`?

File: help/markdown/contributing.md [hunk 4]
```diff
@@ -113,7 +113,11 @@ We [learned from our mistakes](fake-fake5-learn-more.html), so we use the follow
  - For compatibility reasons (migration from legacy). We assume the user doesn't open the global `Fake` namespace.
    
    -> This means we don't add anything in there in the new API.
- - Old APIs are marked as Obsolete with a link (hint) to the new API location.
+ - Old APIs are marked as Obsolete with a link (hint) to the new API location. We use codes to make explicit 
+    - FAKE0001 for moving part from one Module to another
+    - FAKE0002 for removed API we don't know who is using it and how => please open an issue if you use it
+    - FAKE0003 for API that is no more accessible (basically became internal) => please open an issue if you use it
+    - FAKE0004 for API not yet migrated, waiting for your contribution
```
Comment (matthid) [inline help/markdown/contributing.md]: Oh nice idea.

File: help/markdown/core-targets.md [hunk 5]
```diff
@@ -53,27 +53,27 @@ Now we have the following options:
 ## Final targets
 
 Final targets can be used for TearDown functionality. 
-These targets will be executed even if the build fails but have to be activated via ActivateFinalTarget().
+These targets will be executed even if the build fails but have to be activated via ActivateFinal().
```
Comment (matthid) [inline help/markdown/core-targets.md]: Should we write `Target.ActivateFinal` here to advertise the new usage?

File: help/markdown/core-targets.md [hunk 6]
```diff
@@ -53,27 +53,27 @@ Now we have the following options:
...
 
 
 ## Build failure targets
 
 Build failure targets can be used to execute tasks after a build failure.
-These targets will be executed only after a build failure but have to be activated via ActivateBuildFailureTarget().
+These targets will be executed only after a build failure but have to be activated via ActivateBuildFailure().
 
-	BuildFailureTarget "ReportErrorViaMail" (fun _ ->
+	CreateBuildFailure "ReportErrorViaMail" (fun _ ->
```
Comment (matthid) [inline help/markdown/core-targets.md]: `Target.`?

File: help/markdown/core-targets.md [hunk 7]
```diff
@@ -53,27 +53,27 @@ Now we have the following options:
...
 		// send mail about the failure
 	)
 
-	// Activate BuildFailureTarget somewhere during build
-	ActivateBuildFailureTarget "ReportErrorViaMail"
+	// Activate Build Failure Target somewhere during build
+	ActivateBuildFailure "ReportErrorViaMail"
```
Comment (matthid) [inline help/markdown/core-targets.md]: `Target.`?

File: help/markdown/fake-commandline.md [hunk 8]
```diff
@@ -96,24 +96,23 @@ For this short sample we assume you have the latest version of FAKE installed an
...
-	Target "Deploy" (fun () -> trace " --- Deploying app --- ")
+	Target.Create "Deploy" (fun () -> trace " --- Deploying app --- ")
  
  
 	"Clean"
 	  ==> "Build"
 	  ==> "Deploy"
  
-	RunTargetOrDefault "Deploy"
+	Target.RunOrDefault "Deploy"
```
Comment (matthid) [inline help/markdown/fake-commandline.md]: 👍

File: help/markdown/fake-migrate-to-fake-5.md [hunk 9]
```diff
@@ -13,30 +13,35 @@ This "new" version has several advantages:
...
 
-- Regular update to FAKE 5. This should not be breaking. If it breaks you please open an issue.
-- Fix all the (obsolete) warnings in your build-script to use the new API (see the 'Use the new FAKE-API' section).
+- Update to legacy FAKE 5. This should not be breaking. If it breaks you please open an issue.
+
+  - With Paket: add `prerelease` after `nuget FAKE` in paket.dependencies file then `.paket/paket.exe update FAKE`, check that paket.lock references FAKE version 5
```
Comment (matthid) [inline help/markdown/fake-migrate-to-fake-5.md]: We may need to remember to remove `prerelease` after releasing FAKE.

**Comment by matthid** [comment]
Ok I've merged this into https://github.com/fsharp/FAKE/pull/1678 if you want to push additional changes please use a new branch/PR if possible. Thanks.

GitHub should merge/close this PR as soon as https://github.com/fsharp/FAKE/pull/1678 is merged.

File: build-web-bundles.fsx (hunk 3 above)
Comment (matthid) [reply build-web-bundles.fsx]: Nevermind

File: help/markdown/core-targets.md (hunk 5 above)
Comment (matthid) [reply help/markdown/core-targets.md]: fixed.

File: help/markdown/core-targets.md (hunk 6 above)
Comment (matthid) [reply help/markdown/core-targets.md]: fixed.

File: help/markdown/core-targets.md (hunk 7 above)
Comment (matthid) [reply help/markdown/core-targets.md]: fixed.

File: build-web-bundles.fsx (hunk 3 above)
Comment (devcrafting) [reply build-web-bundles.fsx]: Yes I like your suggestion...I will do it and normalize casing in a new PR, do you agree?

Comment (matthid) [reply build-web-bundles.fsx]: actually I was wrong on this. I have no idea where this file is used and if updating this breaks anything...

Comment (devcrafting) [reply build-web-bundles.fsx]: I was thinking you were saying that RunParameterOrDefault would be a better name than just RunOrDefault?

Comment (matthid) [reply build-web-bundles.fsx]: No your change is fine. I'm not sure why we even have runparameterordefault

**Pull request merged**